### PR TITLE
[VEN-949] [VEN-975] Fixes for Certik findings related to liquidations

### PR DESCRIPTION
**Pull request by kkirka**
This PR includes the fixes for the following issues:

* CVP-01 | Improper Checks For `collateralFactor` and `liquidationThreshold` [VEN-949]
* CVP-04 | Possible Insufficient Checks in `liquidateAccount()` and `healAccount()` [VEN-975]

In addition to fixing these issues, the PR adds 100% unit test coverage for batch liquidations and updates the test suite for account healing. There are still some tests missing in VToken and Comptroller hooks, but these are out of the scope of this PR.

### CODE REVIEW

**Review by chechu — COMMENTED** [review]
LGTM

**Comment by chechu** [comment]
From Kirill:
> The problem there is that in the integration tests we’re using the prices from the oracle instead of faked prices, and it now yields “InsufficientShortfall” after I added the check. It should be rather straightforward to fix (just set the faked oracle and set the prices directly)

@UrAvgDeveloper, could you take a look and fix the test?

File: tests/hardhat/Comptroller/accountLiquidityTest.ts [hunk 1]
```diff
@@ -314,53 +314,6 @@ describe("Comptroller", () => {
         .to.be.revertedWithCustomError(comptroller, "MarketAlreadyListed")
         .withArgs(vToken.address);
     });
-
-    it("reverts if not 'in' any markets", async () => {
```
Comment (0xlucian) [inline tests/hardhat/Comptroller/accountLiquidityTest.ts:318]: we do not need this test anymore? Can you elaborate on that?

File: tests/hardhat/Comptroller/healAccountTest.ts [hunk 2]
```diff
@@ -53,23 +50,15 @@ describe("healAccount", () => {
 
     accessControl.isAllowedToCall.returns(true);
     await comptroller.setPriceOracle(oracle.address);
-    await comptroller.setLiquidationIncentive(convertToUnit("1.1", 18));
-    await comptroller.setMinLiquidatableCollateral(convertToUnit("100", 18));
+    await comptroller.setLiquidationIncentive(parseUnits("1.1", 18));
```
Comment (0xlucian) [inline tests/hardhat/Comptroller/healAccountTest.ts:53]: What is the reason for exchaning `convertToUnit`  for `parseUnits`

File: tests/hardhat/Comptroller/accountLiquidityTest.ts (hunk 1 above)
Comment (chechu) [reply tests/hardhat/Comptroller/accountLiquidityTest.ts:318]: https://github.com/VenusProtocol/isolated-pools/pull/167/commits/2e46c045df3cb3b804448b4045277351cf016a17

Comment (0xlucian) [reply tests/hardhat/Comptroller/accountLiquidityTest.ts:318]: Thanks got it

**Review by chechu — APPROVED** [review]

**Pull request merged**